k-distribution/include/builtin/domains.k

```diff
@@ -566,6 +566,7 @@ module STRING
 
   syntax String ::= Float2String ( Float )              [function, functional, hook(STRING.float2string)]
   syntax String ::= Float2String ( Float , String )     [function, klabel(FloatFormat), hook(STRING.floatFormat)]
+  syntax String ::= tokenToString( K )                  [function, hook(STRING.token2string)]
```

Comment:
What is this used for? I would much rather we have a type-safe function instead unless you can demonstrate a use case where we need to be able to transform tokens of any sort via the same function.

Reply:
`#save(Fd, tokenToString(#token("contents", "Bubble"))`. After parsing you may end up with tokens of many types, depending on your grammar. I need a way to connect the contents of those tokens with different primitives like save-to-file, parse-strings...

Reply:
If you just need to convert a token of sort Bubble to a string, you can create a BubbleToString production in your semantics that hooks tto this production. We don't need a production in the prelude for this. I would prefer to keep operations on tokens type safe going forward.